pkg/build/android.go

```diff
@@ -12,13 +12,29 @@ import (
 	"regexp"
 	"time"
 
+	"github.com/google/syzkaller/pkg/config"
 	"github.com/google/syzkaller/pkg/osutil"
 )
 
 type android struct{}
 
+type BuildParams struct {
+	DefconfigFragment string `json:"defconfig_fragment"`
+	BuildTarget       string `json:"build_target"`
+	BuildScript       string `json:"build_script"`
+	VendorBootImage   string `json:"vendor_boot_image"`
+}
+
 var ccCompilerRegexp = regexp.MustCompile(`#define\s+CONFIG_CC_VERSION_TEXT\s+"(.*)"`)
 
+func parseConfig(conf []byte) (*BuildParams, error) {
+	buildCfg := new(BuildParams)
+	if err := config.LoadData(conf, buildCfg); err != nil {
+		return nil, fmt.Errorf("failed to parse build config: %v", err)
+	}
+	return buildCfg, nil
```

Comment:
I think it would be nice to also add some validation rules here (e.g. check whether `BuildScript` is non-empty, etc)

Reply:
Makes sense, thanks! I've added validation rules to check each parameter.